auth/credentials/api.go

```diff
@@ -63,40 +65,58 @@ func NewTokenSourceWithOptions(opts AtlasTokenSourceOptions) TokenSource {
 		ctx = *opts.Context
 	}
 
-	if opts.TokenCache == nil {
-		opts.TokenCache = &InMemoryTokenCache{}
+	config := &clientcredentials.Config{
+		ClientID:     opts.ClientID,
+		ClientSecret: opts.ClientSecret,
+		TokenURL:     tokenURL,
+		AuthStyle:    oauth2.AuthStyleInHeader,
+		Scopes:       []string{}, // No support for scopes.
+	}
+
+	// oauth2 token source for creating tokens
+	tokenSource := config.TokenSource(ctx)
+	if opts.TokenCache != nil {
+		tokenSource = ReuseWrapperCacheTokenSource(tokenSource, opts.TokenCache)
 	}
 
 	return &OAuthTokenSource{
-		clientID:     opts.ClientID,
-		clientSecret: opts.ClientSecret,
-		userAgent:    userAgent,
-		tokenURL:     tokenURL,
-		revokeURL:    revokeUrl,
+		tokenSource:  tokenSource,
 		tokenCache:   opts.TokenCache,
+		revokeURL:    revokeUrl,
+		userAgent:    userAgent,
 		ctx:          ctx,
+		clientID:     opts.ClientID,
+		clientSecret: opts.ClientSecret,
 	}
 }
 
 // NewTokenSource initializes OAuth Token Source that provides a way to obtain valid OAuth Tokens.
 // See credentials.NewTokenSourceWithOptions for advanced use cases.
-func NewTokenSource(clientID, clientSecret string) TokenSource {
+func NewTokenSource(clientID, clientSecret string) oauth2.TokenSource {
 	return NewTokenSourceWithOptions(AtlasTokenSourceOptions{
 		ClientID:     clientID,
 		ClientSecret: clientSecret,
 		UserAgent:    core.DefaultUserAgent,
-		TokenCache:   &InMemoryTokenCache{},
 	})
 }
 
 // NewHTTPClientWithOAuthToken helper method for creating HTTP client with OAuth authentication support.
 // Use this method for performing requests using http.DefaultTransport.
-// For more advanced use cases please create your own credentials.OAuthCustomHTTPTransport.
-func NewHTTPClientWithOAuthToken(client TokenSource) *http.Client {
+// For more advanced use cases, please create your own credentials.OAuthCustomHTTPTransport.
+func NewHTTPClientWithOAuthToken(tokenSource oauth2.TokenSource) *http.Client {
 	return &http.Client{
 		Transport: &OAuthCustomHTTPTransport{
 			UnderlyingTransport: http.DefaultTransport,
-			TokenSource:         client,
+			TokenSource:         tokenSource,
 		},
 	}
 }
+
+// RevokeToken checks if TokenSource is a RevocableTokenSource and revokes Token.
```

Comment:
Since `oauth2.TokenSource` does not have revocation, we're casting `RevocableTokenSource`

Reply:
Actually. I will push a commit that will use RevocableTokenSource as internal TokenSource. We can always revert it.